ios/FluentUI/Vnext/AvatarGroup/AvatarGroup.swift

```diff
@@ -125,36 +125,48 @@ public struct AvatarGroup: View {
-                let avatar = avatars[index]
-                let currentAvatarHasRing = avatar.isRingVisible
-                let nextAvatarHasRing = index + 1 < maxDisplayedAvatars ? avatars[index + 1].isRingVisible : false
-
-                let ringPaddingInterspace = nextAvatarHasRing ? interspace - (ringOffset + ringOuterGap) : interspace - ringOffset
-                let noRingPaddingInterspace = nextAvatarHasRing ? interspace - ringOuterGap : interspace
-                let stackPadding = (currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace)
-
-                let xOrigin = currentAvatarHasRing ? x + ringOffset : x
-                let yOrigin = currentAvatarHasRing ? (nextAvatarHasRing ? ringOuterGap : ringOffset) :
-                    (nextAvatarHasRing ? 0 - ringOffset + tokens.ringOuterGap : 0)
-                let cutoutSize = nextAvatarHasRing ? size + ringOffset + ringOuterGap : size
-
-                AvatarView(avatar)
-                    .modifyIf(needsCutout, { view in
-                        view.mask(AvatarCutout(xOrigin: xOrigin,
-                                               yOrigin: yOrigin,
-                                               cutoutSize: cutoutSize)
-                                    .fill(style: FillStyle(eoFill: true)))
-                    })
-                    .padding(.trailing, tokens.style == .stack ? stackPadding : interspace)
+        let avatarHeight = groupContainsRing() ? CGFloat(size + (ringOffset * 2)) : size
+        GeometryReader { _ in
+            HStack(spacing: 0) {
+                ForEach(0 ..< maxDisplayedAvatars, id: \.self) { index in
+                    // If the avatar is part of Stack style and is not the last avatar in the sequence, create a cutout
+                    let needsCutout = tokens.style == .stack && (overflowCount > 0 || index + 1 < maxDisplayedAvatars)
+                    let avatar = avatars[index]
+                    let currentAvatarHasRing = avatar.isRingVisible
+                    let nextAvatarHasRing = index + 1 < maxDisplayedAvatars ? avatars[index + 1].isRingVisible : false
+
+                    let ringPaddingInterspace = nextAvatarHasRing ? interspace - (ringOffset + ringOuterGap) : interspace - ringOffset
+                    let noRingPaddingInterspace = nextAvatarHasRing ? interspace - ringOuterGap : interspace
+                    let stackPadding = (currentAvatarHasRing ? ringPaddingInterspace : noRingPaddingInterspace)
+
+                    let xOrigin = currentAvatarHasRing ? x + ringOffset : x
+                    let yOrigin = currentAvatarHasRing ? (nextAvatarHasRing ? ringOuterGap : ringOffset) :
+                        (nextAvatarHasRing ? 0 - ringOffset + tokens.ringOuterGap : 0)
+                    let cutoutSize = nextAvatarHasRing ? size + ringOffset + ringOuterGap : size
+                    AvatarView(avatar)
```

Comment:
Provided that the calculations above need to be adjusted, I thought of the following approach:

- not using the GeometryReader
- making the Avatar state implement Identifiable
- using the .offset modifier to position them in the HStack

I think that could improve the animation as well as SwiftUI would keep track of the views an animate them as the state changes:

```
ForEach(avatars.prefix(maxDisplayedAvatars), id: \.id) { avatar in
...
}
```

Reply:
Any findings on implementing Identifiable in the Avatar state?
I'd be curious to see if this wouldn't make the animations smoother.